service/controller/azureconfig/handler/volumebindingmigration/resource.go

```diff
@@ -65,7 +66,13 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	var tenantClusterK8sClient ctrl.Client
 	{
 		tenantClusterK8sClient, err = r.getTenantClusterClient(ctx, &cr)
-		if err != nil {
+		if tenant.IsAPINotAvailable(err) || tenantcluster.IsTimeout(err) {
+			// The kubernetes API is not reachable. This usually happens when a new cluster is being created.
+			// This makes the whole controller to fail and stops next handlers from being executed even if they are
+			// safe to run. We don't want that to happen so we just return and we'll try again during next loop.
+			r.logger.Debugf(ctx, "Kubernetes API is not available, can't proceed.")
```

Comment:
For consistency sake, we could copy/paste the same log lines that we use everywhere else for this scenario
```
tenantClusterK8sClient, err := r.tenantClientFactory.GetClient(ctx, cluster)
	if tenantcluster.IsAPINotAvailableError(err) {
		r.Logger.Debugf(ctx, "tenant API not available yet")
		r.Logger.Debugf(ctx, "canceling resource")

		return nil
	} else if err != nil {
		return microerror.Mask(err)
	}
```

Reply:
yeah makes sense, done